src/array.c

```diff
@@ -96,6 +98,9 @@ enum cc_stat array_new_conf(ArrayConf const * const conf, Array **out)
     ar->mem_alloc  = conf->mem_alloc;
     ar->mem_calloc = conf->mem_calloc;
     ar->mem_free   = conf->mem_free;
+	ar->size       = 0;
+    ar->is_sorted  = false;
+    ar->cmp        = NULL;
```

Comment:
`ar->cmp` should never be set to `NULL`. 

There also doesn't seem to be a way to actually set the `cmp` function. `cmp` should either be passed through `ArrayConf`, in which case there should be a default function that compares pointer values directly, or it should be passed directly to `array_index_of` and the like, through a parameter, which I think would be preferable since it wouldn't be a part of the Array state.